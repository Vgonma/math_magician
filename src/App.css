* {
  box-sizing: border-box;
}

.calculator-grid {
  display: grid;
  margin-top: 2rem;
  grid-template-columns: repeat(4, 6rem);
  grid-template-rows: minmax(5rem, auto) repeat(5, 3.5rem);
}

.output {
  grid-column: 1 / -1;
  background-color: rgba(60, 60, 60, 0.75);
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: flex-end;
}

.previous-operand {
  color: rgba(255, 255, 255, 0.75);
  font-size: 1.5rem;
}

.current-operand {
  color: white;
  font-size: 2.5rem;
}

.span-two {
  grid-column: span 2;
}

.btn {
  border: 1px solid #ccc;
}

.btn:hover {
  background-color: #fff;
}

.btn--orange {
  background-color: rgb(255, 161, 53);
}

.btn--orange:hover {
  background-color: rgb(255, 191, 53);
}

.btn:active {
  border: 2px solid black;
}
