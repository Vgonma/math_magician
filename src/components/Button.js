import React from 'react';

class Button extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render({ sym }) {
    return (
      <button type="button">{sym}</button>
    );
  }
}

export default Button;
