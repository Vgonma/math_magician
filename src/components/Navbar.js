import React, { useState } from 'react';
import { NavLink } from 'react-router-dom';

const Navbar = () => {
  const [active, setActive] = useState('Home');
  return (
    <nav>
      <div>
        <h1>{active}</h1>
      </div>
      <div>
        <NavLink to="/" onClick={() => setActive('Home')}>Home</NavLink>
        <NavLink to="/calculator" onClick={() => setActive('Clculator')}>Calculator</NavLink>
        <NavLink to="/quote" onClick={() => setActive('Quote')}>Quote</NavLink>
      </div>
    </nav>
  );
};

export default Navbar;
