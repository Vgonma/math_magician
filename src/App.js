import { Routes, Route } from 'react-router-dom';
// Component imports
import './App.css';
import Calculator from './components/Calculator';
import Home from './pages/Home';
import Quote from './pages/Quote';
import NotFound from './pages/NotFound';
import Navbar from './components/Navbar';

function App() {
  return (
    <>
      <Navbar />
      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="calculator" element={<Calculator />} />
        <Route path="quote" element={<Quote />} />
        <Route path="*" element={<NotFound />} />
      </Routes>
    </>
  );
}

export default App;
