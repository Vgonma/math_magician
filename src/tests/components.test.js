import { BrowserRouter } from 'react-router-dom';
import {
  render,
  screen,
  fireEvent,
} from '@testing-library/react';
import renderer from 'react-test-renderer';
import Navbar from '../components/Navbar';
import Button from '../components/Button';
import App from '../App';

describe('Components render correctly', () => {
  it('Navbar component', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Navbar interaction', () => {
    render(<BrowserRouter><App /></BrowserRouter>);
    fireEvent.click(screen.getByText('Quote'));
    expect(screen.getByText('Random Quote')).toBeInTheDocument();
  });

  it('calculator interaction', () => {
    render(<BrowserRouter><App /></BrowserRouter>);
    fireEvent.click(screen.getByText('Calculator'));
    fireEvent.click(screen.getByText('5'));
    fireEvent.click(screen.getByText('9'));
    expect(screen.getByText('59')).toBeInTheDocument();
  });

  it('Button component', () => {
    const tree = renderer.create(
      <Button sym="AC" className="btn" clickHandler={() => {}} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
