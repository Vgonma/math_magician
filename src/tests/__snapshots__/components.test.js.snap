// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components render correctly Button component 1`] = `
<button
  className="btn"
  onClick={[Function]}
  type="button"
>
  AC
</button>
`;

exports[`Components render correctly Navbar component 1`] = `
<nav
  className="navbar"
>
  <div
    className="page-title"
  >
    <h1>
      Home
    </h1>
  </div>
  <div
    className="navlinks-container"
  >
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      className=""
      href="/calculator"
      onClick={[Function]}
    >
      Calculator
    </a>
    <a
      className=""
      href="/quote"
      onClick={[Function]}
    >
      Quote
    </a>
  </div>
</nav>
`;
